src/include/homestore/chunk_selector.h

```diff
@@ -21,8 +21,8 @@ class Chunk;
 class ChunkSelector {
 public:
     ChunkSelector() = default;
-    virtual void add_chunk(cshared< Chunk >&) = 0;
-    virtual void foreach_chunks(std::function< void(cshared< Chunk >&) >&& cb) = 0;
+    virtual void add_chunk(cshared< Chunk >) = 0;
+    virtual void foreach_chunks(std::function< void(cshared< Chunk >) >&& cb) = 0;
```

Comment:
I am not finding the reason to change Chunk to Chunk&. It puts onus on the caller to move and if they are using it for only referencing we should pass Chunk&. Sending shared_ptr by value for referencing is not the right thing to do.

Reply:
sorry for this , i just push a PR w/o cancel the test modification, will change it back